npm-publish/bin/publish.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+
+set -o errexit   # exit on error
+set -o errtrace  # exit on error within function/sub-shell
+set -o nounset   # error on undefined vars
+set -o pipefail  # error if piped command fails
+
+# Default variables
+RELEASE_TYPE=
+MAIN_BRANCH="trunk"
+
+echo_title() {
+	echo ""
+	echo "== $1 =="
+}
+
+while getopts ":t:" option;
+do
+	case $option in
+		# npm major/minor/patch
+		t) RELEASE_TYPE=$OPTARG ;;
+
+		\?) echo "Error: Invalid param / option specified"
+			exit 199 ;;
+   esac
+done
+
+# Validate release type value
+if [ "$RELEASE_TYPE" != "major" ] && [ "$RELEASE_TYPE" != "minor" ] && [ "$RELEASE_TYPE" != "patch" ]; then
+	echo "❌ Invalid release type specified. Please make sure the -t flag is one of major/minor/patch."
+	exit 200
+fi
+
+# Fetch some basic package information
+echo_title "Fetching local package info"
+LOCAL_NAME=$(node -p "require('./package.json').name")
+LOCAL_VERSION=$(node -p "require('./package.json').version")
+LOCAL_BRANCH=$(git branch --show-current)
+REMOTE_VERSION=$(npm view "$LOCAL_NAME" version)
+echo "✅ Found $LOCAL_NAME $LOCAL_VERSION on branch $LOCAL_BRANCH"
+echo "✅ Published version is $REMOTE_VERSION"
+echo "✅ Will publish new $RELEASE_TYPE release"
+
+# Validate npm is logged in and ready
+echo_title "Checking npm auth"
+if ! NPM_USER=$( npm whoami ); then
```

Comment:
Where are we doing `npm login` or such?

Reply:
NPM_TOKEN is added as a secret in the repo using thos and then passed into the workflow as an input.